### PR TITLE
Remove qwery and bonzo from article-aside-adverts.js and convert it to TypeScript

### PR DESCRIPTION
**Pull request by marsavar**
## What does this change?
This change removes `qwery` and `bonzo` from `article-aside-adverts.js` and convers the module to TypeScript.
The module removes sticky behaviour of ads in the right column from articles with immersive elements.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?
Reducing the bundle size and removing deprecated libraries is good. Hard to say how valuable this is, considering frontend-rendered immersive articles are going to be deprecated soon.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [x] Yes

The vast majority of immersive articles are rendered by DCR, but there are still a few isolated cases where they're rendered by frontend, namely GLabs articles (example [here](https://www.theguardian.com/saving-for-a-sunny-day-with-nsandi/2021/apr/20/its-incredibly-liberating-what-saving-for-a-piano-taught-me-about-my-finances)).

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Review by arelra — APPROVED** [review]
Nice job on a tough conversion!

File: static/src/javascripts/projects/commercial/modules/article-aside-adverts.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+const minArticleHeight = 1300;
+
+const config = config_ as {
+	get: (s: string, b?: boolean) => string;
+};
+
+const getAllowedSizesForImmersive = (availableSpace: number) => {
+	// filter ad slot sizes based on the available height
+	// mark: 01303e88-ef1f-462d-9b6e-242419435cec
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/article-aside-adverts.ts:14]: This usually indicates that there's a equivalent file elsewhere (in this case DCR):

https://github.com/search?q=org%3Aguardian+01303e88-ef1f-462d-9b6e-242419435cec&type=code

**Review by mxdvl — CHANGES_REQUESTED** [review]
Well done on the conversion! I'm happy for this to go through with two small caveats:

1. We need to make sure that this does not break parity with DCR, as mentioned above, by checking the matching `mark` &rarr; [`AdSlot.tsx`](https://github.com/guardian/dotcom-rendering/blob/4326928bd2c8285eeb1db5303e29e8e13a09a1c4/src/web/components/AdSlot.tsx).
2. When converting to TypeScript, it's worth removing the javascript.flow.archive equivalent of the JS original. In this case:

  - https://github.com/guardian/frontend/blob/main/static/src/javascripts.flow.archive/projects/commercial/modules/article-aside-adverts.js
  - https://github.com/guardian/frontend/blob/main/static/src/javascripts.flow.archive/projects/commercial/modules/article-aside-adverts.spec.js

**Comment by MarSavar** [comment]
Thank you Max, useful tips! Will do that asap.

**Review by mxdvl — APPROVED** [review]

**Pull request merged**